rasa/nlu/data_augmentation/augmenter.py

```diff
@@ -0,0 +1,699 @@
+        classification_report: Classification report of the model run *without* data augmentation.
+        training_intents: All intents in the training data (non-augmented).
+        pooled_intents: The intents that have been selected for data augmentation.
+        output_directory: Directory to store the output reports in.
+
+    Returns:
+        Tuple representing the data augmentation summary as well as the set of changed intents.
+    """
+
+    # Retrieve intents for which performance has changed
+    changed_intents = (
+        _get_intents_with_performance_changes(
+            classification_report, intent_report, training_intents,
+        )
+        - pooled_intents
+    )
+
+    # Create and update result reports
+    report_tuple = _create_augmentation_summary(
+        pooled_intents, changed_intents, classification_report, intent_report,
+    )
+
+    intent_summary = report_tuple[0]
+    intent_report.update(report_tuple[1])
+
+    # Store report to file
+    rasa.shared.utils.io.dump_obj_as_json_to_file(
+        os.path.join(output_directory, "intent_report.json"), intent_report,
+    )
+
+    return intent_summary
+
+
+def _plot_summary_report(
+    intent_summary: Dict[Text, Dict[Text, float]], output_directory: Text,
+) -> None:
+    """Plots the data augmentation summary.
+
+    Args:
+        intent_summary: Summary report of the effect of data augmentation.
+        output_directory: Directory to store the summary plot in.
+    """
+    for metric in ["precision", "recall", "f1-score"]:
+        output_file = os.path.join(output_directory, f"{metric}_changes.png")
+        rasa.utils.plotting.plot_intent_augmentation_summary(
+            augmentation_summary=intent_summary, metric=metric, output_file=output_file,
+        )
+
+
+def _get_intents_with_performance_changes(
+    classification_report: Dict[Text, Dict[Text, Any]],
+    intent_report: Dict[Text, float],
+    all_intents: List[Text],
+    significant_figures: int = 2,
+) -> Set[Text]:
+    """
+    Extracts the intents whose performance has changed.
+
+    Args:
+        classification_report: Classification report of the model run *without* data augmentation.
+        intent_report: Report of the model run *with* data augmentation.
+        all_intents: List of all intents.
+        significant_figures: Significant figures to be taken into account when assessing whether the performance of an
+            intent has changed.
+
+    Returns:
+        Set of intents that have changed - i.e. that were affected by data augmentation.
+    """
+    changed_intents = set()
+    for intent_key in all_intents:
+        for metric in ["precision", "recall", "f1-score"]:
+            rounded_original = round(
+                classification_report[intent_key][metric], significant_figures
+            )
+            rounded_augmented = round(
+                intent_report[intent_key][metric], significant_figures
```

Comment:
Please use `.get()` to access keys of dictionary to keep it consistent across codebase. If you view in pycharm, there should be a warning indicating that it's an anti-pattern. 😉 
Please change it anywhere else applicable in the PR.